pkg/kv/dynamodb/store.go

```diff
@@ -356,6 +357,9 @@ func (s *Store) scanInternal(ctx context.Context, partitionKey, scanKey []byte,
 	if s.params.ScanLimit != 0 {
 		queryInput.SetLimit(s.params.ScanLimit)
 	}
+	if options.BatchSize != 0 {
+		queryInput.Limit = aws.Int64(int64(options.BatchSize))
+	}
```

Comment:
We should choose the minimum between ScanLimit and batchSize

Reply:
Fixing